src/electron/future/services/vector-store.ts

```diff
@@ -0,0 +1,311 @@
+	 * the service is fully operational for subsequent operations.
+	 *
+	 * @throws {Error} If the Qdrant binary is not found at the specified path.
+	 * @throws {Error} If Qdrant can't be started.
+	 */
+	private async start() {
+		if (!this.process) {
+			const cwdPath = getCaptainData("qdrant");
+			const binaryPath = getCaptainData("qdrant/qdrant.exe");
+			const configPath = getCaptainData("qdrant/config.yaml");
+
+			const scriptArguments = ["--config-path", configPath];
+
+			// Check if the binary exists
+			try {
+				await fsp.access(binaryPath);
+			} catch {
+				throw new Error(`Qdrant binary not found at path: ${binaryPath}`);
+			}
+
+			// Start the process and wait until its ready
+			try {
+				this.process = execa(binaryPath, [...scriptArguments], { cwd: cwdPath });
+
+				await this.serviceReady();
+			} catch (error) {
+				throw new Error(`Qdrant can't be started: ${error}`);
+			}
+		}
+	}
+
+	/**
+	 * Stops the Qdrant service if it is running.
+	 */
+	public async stop() {
+		if (this.process) {
+			this.process.kill();
+			this.process = undefined;
+		}
+	}
+
+	/**
+	 * Initializes the Qdrant client for communicating with the Qdrant service.
+	 */
+	private initializeClient() {
+		if (!this.client) {
+			this.client = new QdrantClient({ url: this.url });
+		}
+	}
+
+	/**
+	 * Waits for the Qdrant service to become fully operational by repeatedly checking its
+	 * health endpoint.
+	 *
+	 * @param {ServiceReadyConfig} config - `maxRetries`, `retryInterval`, and `timeout`
+	 * @throws {Error} If the Qdrant service does not become ready within the specified retries and interval.
+	 */
+	private async serviceReady(config: ServiceReadyConfig = {}) {
+		const { maxRetries = 30, retryInterval = 1000, timeout = 5000 } = config;
+
+		const healthEndpoint = `${this.url}`;
+
+		for (let index = 0; index < maxRetries; index++) {
+			try {
+				const response = await axios.get(healthEndpoint, { timeout });
+
+				if (response.status === 200) {
+					console.log("Qdrant is ready.");
+					return;
+				}
+			} catch {}
+
+			await new Promise(resolve => {
+				setTimeout(resolve, retryInterval);
+			});
+		}
+
+		throw new Error("Qdrant service did not become ready in time.");
+	}
+
+	/**
+	 * Returns the current instance of the VectorStore class.
+	 *
+	 * @returns {VectorStore} The current VectorStore instance.
+	 * @throws {Error} If the instance has not been initialized yet.
+	 */
+	public static get getInstance(): VectorStore {
+		if (!VectorStore.instance) {
+			throw new Error(
+				"VectorStore instance has not been initialized. Please call 'init' first."
+			);
+		}
+
+		return VectorStore.instance;
+	}
+
+	/**
+	 * Inserts or updates documents in the specified collection. If a document ID is not provided,
+	 * the method will attempt to find an existing document based on the payload 'id' and 'language'.
+	 * If an existing document is not found, a new UUID will be generated.
+	 * 
+	 * Example: Auto-ID (recommended)
```

Comment:
example snippets should use the `@example` directive

Reply:
Thanks, those have been fixed.